Docker-Compose Configuration

Aersaud opened the pull request:
This repo can be cloned into a folder and once the config/docker-compose files are edited can be run with docker-compose up -d. This will automatically set up the MariaDB server and the sync container. 

The sync container is run under user syncuser at  /home/syncuser/sync
The MariaDB container is set up with environment variables via docker-compose to set up the database. All MariaDB files are by default stores in the same folder as the cloned repo in ./mysql.

This was tested in production with Nginx as a reverse proxy and is fully functional ONCE the files are edited to YOUR specific configuration.

Aersaud committed:
Merge with upstream

Update dockerfile

Update README.md

Delete config.docker.yaml

Delete docker-entrypoint.sh

Delete run.sh

Create config.yaml

Create docker-compose.yml

Update docker-compose.yml for readability, create get_latest_configs.sh to get the latest configuration files, recreated .gitignore, update README.md

Squash and Rebase with upstream - updated to docker-compose, update dockerfile, update readme, added script to get lastest configs from sync repo, created docker-compose.yml

Added information for reverse proxy, and nginx example

Revert "Added information for reverse proxy, and nginx example"

This reverts commit 8394e968078aded38df99a61e32dd9d47c052381.

Revert "Update docker-compose.yml for readability, create get_latest_configs.sh to get the latest configuration files, recreated .gitignore, update README.md"

This reverts commit d1b6bb3b7dd834b3aed6b9e510e8b58cb57a5c03.

Revert "Create docker-compose.yml"

This reverts commit 6f37588c2183e479987bc8456c5f2e684f74db4e.

Revert "Create config.yaml"

This reverts commit e52a21d2f19254fe4d2f0f70be27b97a605fdbb9.

Revert "Delete run.sh"

This reverts commit 1e970de12d8c780ab6a90ba2f137bfec50eeb6d3.

Revert "Delete docker-entrypoint.sh"

This reverts commit 798ced5949b2c06e216240f6e9cc3082e97df700.

Revert "Delete config.docker.yaml"

This reverts commit 9efaf9737f2a6e257353e949ab785d87f8e8ad71.

Revert "Update README.md"

This reverts commit 578ef68bfb1ecead7eca82cb713f4da8e43f2bee.

Revert "Update dockerfile"

This reverts commit 4ee67bad4aa253948badfa75786b9ca9d0f0b54a.

Revert "Delete .gitignore"

This reverts commit c94f64dad899fca994b9694f55b553aecb56d449.

Revert "Squash and Rebase with upstream - updated to docker-compose, update dockerfile, update readme, added script to get lastest configs from sync repo, created docker-compose.yml"

This reverts commit f853f411053c7033805e51d5e74093d8aa0f1929.

Merge to master

Aersaud committed:
Delete .gitignore

Aersaud committed:
Update dockerfile

Aersaud committed:
Update README.md

Aersaud committed:
Delete config.docker.yaml

Aersaud committed:
Delete docker-entrypoint.sh

Aersaud committed:
Delete run.sh

Aersaud committed:
Create config.yaml

Aersaud committed:
Create docker-compose.yml

calzoneman commented on .gitignore:
```diff
@@ -1,2 +0,0 @@
-.DS_Store
```
Can we keep this `.gitignore`?

calzoneman commented on Dockerfile:
```diff
@@ -1,72 +1,16 @@
-FROM alpine:3.8
+from node:latest
```
Is it possible to pin it to a specific major version (e.g. 12.x)?  I generally test CyTube against the current LTS version of node; sometimes the newer versions introduce regressions and/or incompatibilities with certain C++ modules that then need to be updated to support it.

calzoneman commented on README.md:
```diff
@@ -1,77 +1,8 @@
+https://github.com/calzoneman/sync
 
-*This repository is currently very alpha and a work in progress*
+Clone the repo, edit the config file with your information. Make sure the docker-compose file matches your config where needed (port numbers, passwords, certs)
```
Could you add a basic getting started section like the previous README had, with example commands for building the image and starting it with docker-compose?

calzoneman commented on config.yaml:
```diff
@@ -4,11 +4,11 @@
+  database: 'cytube3'
+  user: 'cytube3'
+  password: 'super_secure_password'
   pool-size: 10
```
Since you're mounting in the config anyways, would it make sense to ask people to get the config template from upstream so that you don't have to update this one whenever the main repo changes it?  There is also a separate `conf` folder with auxiliary configuration.

calzoneman commented on docker-compose.yml:
```diff
@@ -0,0 +1,37 @@
+     depends_on:
+       - "mysql"
+     ports:
+# CHANGE TO MATCH YOUR CONFIG/PROXY
```
It's a minor thing but could you capitalize these comments normally instead of all caps?

calzoneman commented on docker-compose.yml:
```diff
@@ -0,0 +1,37 @@
+       - "./certs:/etc/certs"
+# CHANGE TO IMPORT YOUR CONFIG FILE
+       - "./config.yaml:/home/syncuser/sync/config.yaml"
+#  UNCOMMENT IF YOU PLAN ON USING A REVERSE PROXY - YOU MUST UPDATE THIS TO MATCH YOUR PROXY NETWORK       
```
I'm not sure what "proxy network" means, if it's expected that most people using this image would want a reverse proxy you could just include it by default, otherwise it might be helpful to provide an example of how it works or just exclude it if it's specific to your setup.

calzoneman reviewed (commented):
Could you rebase/squash these commits into one?  https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing

Aersaud commented:
I will work on these over this weekend :)

Aersaud committed:
Update docker-compose.yml for readability, create get_latest_configs.sh to get the latest configuration files, recreated .gitignore, update README.md

Aersaud committed:
Added information for reverse proxy, and nginx example

Aersaud committed:
Squash and Rebase with upstream - updated to docker-compose, update dockerfile, update readme, added script to get lastest configs from sync repo, created docker-compose.yml

Added information for reverse proxy, and nginx example

Revert "Added information for reverse proxy, and nginx example"

This reverts commit 8394e968078aded38df99a61e32dd9d47c052381.

Revert "Update docker-compose.yml for readability, create get_latest_configs.sh to get the latest configuration files, recreated .gitignore, update README.md"

This reverts commit d1b6bb3b7dd834b3aed6b9e510e8b58cb57a5c03.

Revert "Create docker-compose.yml"

This reverts commit 6f37588c2183e479987bc8456c5f2e684f74db4e.

Revert "Create config.yaml"

This reverts commit e52a21d2f19254fe4d2f0f70be27b97a605fdbb9.

Revert "Delete run.sh"

This reverts commit 1e970de12d8c780ab6a90ba2f137bfec50eeb6d3.

Revert "Delete docker-entrypoint.sh"

This reverts commit 798ced5949b2c06e216240f6e9cc3082e97df700.

Revert "Delete config.docker.yaml"

This reverts commit 9efaf9737f2a6e257353e949ab785d87f8e8ad71.

Revert "Update README.md"

This reverts commit 578ef68bfb1ecead7eca82cb713f4da8e43f2bee.

Revert "Update dockerfile"

This reverts commit 4ee67bad4aa253948badfa75786b9ca9d0f0b54a.

Revert "Delete .gitignore"

This reverts commit c94f64dad899fca994b9694f55b553aecb56d449.

Gelmo commented:
I'll take a look at this as well, tomorrow. I was planning on renewing the image over the next week, so great timing!

Aersaud committed:
Revert "Added information for reverse proxy, and nginx example"

This reverts commit 8394e968078aded38df99a61e32dd9d47c052381.

Aersaud committed:
Revert "Update docker-compose.yml for readability, create get_latest_configs.sh to get the latest configuration files, recreated .gitignore, update README.md"

This reverts commit d1b6bb3b7dd834b3aed6b9e510e8b58cb57a5c03.

Aersaud committed:
Revert "Create docker-compose.yml"

This reverts commit 6f37588c2183e479987bc8456c5f2e684f74db4e.

Aersaud committed:
Revert "Update dockerfile"

This reverts commit 4ee67bad4aa253948badfa75786b9ca9d0f0b54a.

Aersaud committed:
Revert "Delete config.docker.yaml"

This reverts commit 9efaf9737f2a6e257353e949ab785d87f8e8ad71.

Aersaud committed:
Revert "Update README.md"

This reverts commit 578ef68bfb1ecead7eca82cb713f4da8e43f2bee.

Aersaud committed:
Revert "Create config.yaml"

This reverts commit e52a21d2f19254fe4d2f0f70be27b97a605fdbb9.

Aersaud committed:
Revert "Delete run.sh"

This reverts commit 1e970de12d8c780ab6a90ba2f137bfec50eeb6d3.

Aersaud committed:
Revert "Delete docker-entrypoint.sh"

This reverts commit 798ced5949b2c06e216240f6e9cc3082e97df700.

Aersaud committed:
Revert "Delete .gitignore"

This reverts commit c94f64dad899fca994b9694f55b553aecb56d449.

Aersaud committed:
Revert "Revert "Delete .gitignore""

This reverts commit 6280a5db22cf89edc899428c13c3766686f98181.

Aersaud committed:
Revert "Revert "Update dockerfile""

This reverts commit 082b6368405974d85148a1b363517d8f5a026ddb.

Aersaud committed:
Revert "Revert "Delete config.docker.yaml""

This reverts commit 099c01a575f802b3d7214ebee6dd64844064cd94.

Aersaud committed:
Revert "Revert "Update README.md""

This reverts commit 7c78086a1df619c59d4d1964a58612e9626b6525.

Aersaud committed:
Revert "Revert "Delete docker-entrypoint.sh""

This reverts commit fe59b0fa8102ca787db16afb161455906d2e37a0.

Aersaud committed:
Revert "Revert "Delete run.sh""

This reverts commit dc9641ce7e256632aaeb065af915ca35f9ad0cd0.

Aersaud committed:
Revert "Revert "Create config.yaml""

This reverts commit c005c1c2ccb884db4157e4ae494f76e04ffae51f.

Aersaud committed:
Revert "Revert "Create docker-compose.yml""

This reverts commit 86a008dd8abe19654ba6077fd786ab886aff2730.

Aersaud committed:
Revert "Revert "Update docker-compose.yml for readability, create get_latest_configs.sh to get the latest configuration files, recreated .gitignore, update README.md""

This reverts commit 5c3982278ae20bd2ab7132ac333e5e5f7caa0b0f.

Aersaud committed:
Revert "Squash and Rebase with upstream - updated to docker-compose, update dockerfile, update readme, added script to get lastest configs from sync repo, created docker-compose.yml"

This reverts commit f853f411053c7033805e51d5e74093d8aa0f1929.

Aersaud commented:
Not so sure I am doing the rebase and squash correctly *facepalm* I updated everything else with more documentation, added a way to get the latest configs, and added an installation step by step.  @calzoneman

Aersaud committed:
Update readme

Aersaud committed:
Merge to master

Aersaud committed:
Merge branch 'master' of https://github.com/Aersaud/cytube-docker

calzoneman commented:
Hey, sorry I haven't had time to review this recently.  I will probably get to it this week.  @Gelmo do you still plan to review?

Gelmo commented:
@calzoneman Yea, just been busier than expected. I'm hoping to review and test on Thursday night.

calzoneman commented on README.md:
````diff
@@ -1,77 +1,129 @@
-MIXER_ID=null
-GIT_COMMIT=0bc866dbfaba0ef09dc703d6ef261bdf2e70b2ff
-```
+Next you will need to run get_latest_configs.sh in order to automatically pull the latest versions of the config from the sync repo. You may also choose to do this manually. You will need the conf/example and a config.yaml files so that they can be mounted into the container. The script is there to make this easier for you by downloading the necessary files.
````
I'm kind of wondering if it makes more sense to add calzoneman/sync as a [git submodule](https://github.blog/2016-02-01-working-with-submodules/) of this repo so that everything gets checked out with one `git clone && git submodule init && git submodule update`

Aersaud replied:
Added https://github.com/calzoneman/sync.git as a submodule

calzoneman commented on README.md:
```diff
@@ -1,77 +1,129 @@
+calzoneman has thoroughly commented the file to give you a good idea of what needs to be edited.
 
-https://hub.docker.com/_/mariadb/
+After editting your config.yaml file you will need to edit the docker-compose.yml to match what you have entered in the config.yaml file. It is commented to give you an idea of what needs to be editted.
```
Typo: editing/edited

Aersaud replied:
Fixed typos in the readme. Thanks I didn't notice it :smile:

calzoneman commented on README.md:
````diff
@@ -1,77 +1,129 @@
+The file has comments to help you edit what is necessary.
 ```
-docker build -t cytube-docker .
+version: "3.7"
````
Is this the same file that is also saved as `docker-compose.yml` in this pull request?  If so I think we can just keep that commented one and remove the one from README.md for clarity (and to avoid them getting out of sync)

Aersaud replied:
Removed this from the readme for clarity.

calzoneman commented on get_latest_configs.sh:
```diff
@@ -0,0 +1,14 @@
+#bin/bash
+
+wget -q -O config.template.yaml https://raw.githubusercontent.com/calzoneman/sync/3.0/config.template.yaml
```
If we don't end up going the submodule route, we should add a check here in case the user doesn't have `wget` installed.  Or better yet, try `wget` and `curl` and if neither is installed, prompt the user to install files manually.  Something like this:

```sh
#!/bin/sh

if hash wget 2>/dev/null
then
  DOWNLOAD_CMD="wget -q -O"
elif hash curl 2>/dev/null
  DOWNLOAD_CMD="curl -o"
else
  echo "Could not find wget or curl, please download configuration file manually from https://github.com/calzoneman/sync" >&2
  exit 1
fi
```

Aersaud replied:
I changed the way this file works since I added sync as a submodule. Instead it will be copying the files from the sync folder, rather than downloading them.

calzoneman commented on get_latest_configs.sh:
```diff
@@ -0,0 +1,14 @@
+#bin/bash
```
Missing the `!` for `#!/bin/bash`, also it'd be nice to just use POSIX shell (`/bin/sh`) in case someone is using zsh instead of bash -- looks like this script doesn't do anything super fancy so we shouldn't need to worry about bash-specific features.

Aersaud replied:
Updated to use ```#!/bin/sh```

calzoneman commented on get_latest_configs.sh:
```diff
@@ -0,0 +1,14 @@
+
+wget -q -O config.template.yaml https://raw.githubusercontent.com/calzoneman/sync/3.0/config.template.yaml
+mv config.template.yaml config.yaml
+echo -e "\e[32m Success! Config file downloaded from github and copied to config.yaml. Please update the config to match the settings specific to your application."
```
Without checking the return code of `wget` you don't necessarily know it succeeded.  The easiest way to avoid this is to add `set -e` to the top of the script (below the `#!/bin/sh`) to crash the shell script if any command fails.

Aersaud replied:
Thanks for the tip! I will  use this in the future with other projects. This is not needed now since it will be copying the files directly from the sync folder which is not a git submodule.

Aersaud committed:
Update readme, removed docker-compose.yml from read me

Aersaud committed:
Edit and rename get_latest_configs.sh to copy_config.sh

Aersaud committed:
Added sync as submodule

Aersaud committed:
Update readme

Aersaud committed:
Update submodule + readme

Aersaud committed:
Update from working master

Aersaud committed:
Merge branch 'master' of https://github.com/Aersaud/cytube-docker into upstream

Aersaud commented:
Updated the master branch with the requested changes.

Aersaud committed:
Fix typos in README

Aersaud committed:
Merge pull request #1 from Aersaud/upstream

Merge README from upstream to master

Aersaud committed:
Update readme to specify repo must be cloned recursivly

Aersaud committed:
Merge pull request #2 from Aersaud/upstream

Update readme to specify repo must be cloned recursivly

Aersaud committed:
Cleaned up readme. There was a copy paste error

Aersaud committed:
Merge pull request #3 from Aersaud/upstream

Cleaned up readme. There was a copy paste error

calzoneman commented:
@Gelmo any thoughts?

Gelmo commented:
Sorry for the delays here, been a brutal work week (shifting to work at home due to Covid-19). I will do my best to review this later tonight

Gelmo commented:
Didn't have an opportunity tonight, so I won't be able to review until Saturday night

calzoneman commented:
I'm merging, Gelmo if you have thoughts feel free to submit a PR to fix it later.
